src/Middleware/HttpLogging/src/W3CLoggerProcessor.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.HttpLogging
+{
+    internal class W3CLoggerProcessor : FileLoggerProcessor
+    {
+        private readonly W3CLoggingFields _loggingFields;
+
+        public W3CLoggerProcessor(IOptionsMonitor<W3CLoggerOptions> options, IHostEnvironment environment, ILoggerFactory factory) : base(options, environment, factory)
+        {
+            _loggingFields = options.CurrentValue.LoggingFields;
+        }
+
+        public override async Task OnFirstWrite(StreamWriter streamWriter)
+        {
+            await WriteMessageAsync("#Version: 1.0", streamWriter);
+
+            await WriteMessageAsync("#Start-Date: " + DateTimeOffset.Now.ToString("yyyy-MM-dd HH:mm:ss", CultureInfo.InvariantCulture), streamWriter);
```

Comment:
Should this be UTC?

Reply:
It makes more sense to me to have it in the user's local time, do you think there's a strong reason for using UTC?

Reply:
>Times are recorded in the form HH:MM, HH:MM:SS or HH:MM:SS.S where HH is the hour in 24 hour format, MM is minutes and SS is seconds. All times are specified in GMT.

From the W3C format documentation